lms/envs/production.py

```diff
@@ -527,6 +527,7 @@ def get_env_setting(setting):
 BROKER_TRANSPORT_OPTIONS = {
     'fanout_patterns': True,
     'fanout_prefix': True,
+    **ENV_TOKENS.get('CELERY_BROKER_TRANSPORT_OPTIONS', {})
```

Comment:
@gabor-boros, should we catch `TypeError` here and raise `ImproperlyConfigured` instead, if a user provides something else than a dict?

It might be safer to move it out to edx/configurations. This could be overkill, though.
cc: @pomegranited

Reply:
@Agrendalath I believe it would be an overkill to do.

Reply:
@Agrendalath  Yeah, since edx/configuration is being deprecated, it's better to add this to the `ENV_TOKENS`.

But I do think catching `ImproperlyConfigured` would be sensible.

Reply:
@pomegranited 

> But I do think catching ImproperlyConfigured would be sensible.

What @Agrendalath proposed is to **raise** an `ImproperlyConfigured` error, not catching that. Or is that caught somewhere? If we would do that, it would seem like this:

```python
try:
  BROKER_TRANSPORT_OPTIONS = {
      'fanout_patterns': True,
      'fanout_prefix': True,
      **ENV_TOKENS.get('CELERY_BROKER_TRANSPORT_OPTIONS', {})
  }
except TypeError as exc:
  raise ImproperlyConfigured('CELERY_BROKER_TRANSPORT_OPTIONS') from exc
```

Would you like me to update the PR with this?

Reply:
@gabor-boros  Sorry yes -- raising `ImproperlyConfigured` is better than getting a `TypeError`; your proposed change looks good.